src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -86,21 +84,38 @@ public CommonCompiler(CommandLineParser parser, string responseFile, string[] ar
 
         internal abstract bool SuppressDefaultResponseFile(IEnumerable<string> args);
 
-        internal string GetAssemblyFileVersion()
+        /// <summary>
+        /// The type of the compiler class for version information in /help and /version.
+        /// We don't simply use this.GetType() because that would break mock subclasses.
+        /// </summary>
+        internal abstract Type Type { get; }
```

Comment:
NIT: I'd call it CompilerVersionType  or something like that. "Type" is already pretty overloaded term when code is in a compiler and is also using reflection.